addons/account_fleet/models/account_move.py

```diff
@@ -59,3 +60,15 @@ def write(self, vals):
     def unlink(self):
         self.sudo().vehicle_log_service_ids.with_context(ignore_linked_bill_constraint=True).unlink()
         return super().unlink()
+
+    @api.model
+    def _get_extra_query_base_tax_line_mapping(self) -> SQL:
+        """Override to add vehicle_id matching condition for tax details query.
+        This ensures that tax lines are only matched with base lines that have the same vehicle_id,
+        preventing tax lines from being incorrectly merged when different vehicles are used.
+        """
```

Comment:
This will replace any other `_get_extra_query_base_tax_line_mapping` override, but we should keep it